src/lib.rs

```diff
@@ -125,8 +124,8 @@ fn json_set(ctx: &Context, args: Vec<String>) -> RedisResult {
     match (current, set_option) {
         (Some(ref mut doc), ref op) => {
             if doc.set_value(&value, &path, op, format)? {
-                if let Some(index) = index {
-                    index::add_document(&key, &index, &doc)?;
+                if let Some(index) = value_index {
```

Comment:
For consistency this should be `if let Some(value_index) = value_index {`

Reply:
:+1: